src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -110,23 +111,68 @@ public static void closeExpiredAndIdleConnections() {
     }
   }
 
-  public static void setProxyForS3(ClientConfiguration clientConfig) {
-    if (useProxy) {
-      clientConfig.setProxyHost(proxyHost);
-      clientConfig.setProxyPort(proxyPort);
-      clientConfig.setNonProxyHosts(nonProxyHosts);
-      if (!Strings.isNullOrEmpty(proxyUser) && !Strings.isNullOrEmpty(proxyPassword)) {
-        clientConfig.setProxyUsername(proxyUser);
-        clientConfig.setProxyPassword(proxyPassword);
+  public static void setS3ProxyFromProperties(
```

Comment:
Instead of using static proxy values (which no longer exist), use the properties given.